### ISSUE TITLE
Copy to clipboard not working in 2.1.15.17101701 and Firefox ESR 52.x

### ISSUE THREAD

**Issue by 48kRAM** [nuxsmin/sysPass#820]
Hello,

Just upgraded from 1.2.x to 2.1.15.17101701. On the upgraded version, clicking the clipboard icon to copy the password results in a red error banner:

"Error while copying to clipboard"

I'm able to view the password though. Verified that the error occurs on simple alphanumeric passwords.

Server is RedHat Enterprise 6. PHP is 5.6.30

**Comment by 48kRAM** [comment]
Bug is also present when using Chrome 63 as client so I've edited the issue title.

**Comment by nuxsmin** [comment]
Hello @48kRAM  did you try to show the password dialog and then copy it to the clipboard?

The key difference between 2.1 and 1.2 version is a performance issue, since 2.1 implements a stronger encryption mechanism which makes password copy requests take longer time to send the response., thus the browser breaks the response because security reasons.

There is an open issue regarding this behaviour.

Regards

**Comment by 48kRAM** [comment]
Yes - I can show the password and then copy it to the clipboard. I'll let my folks know to use this workaround for now.

**Comment by vmario89** [comment]
Hi,
my colleague has got the same problem that sometimes the error occurres that he cannot copy password to clipboard. He uses current Chrome. I am using FireFox for myself where the error did not occurre until now. Do you have an idea how to solve this? Whats the number of the ticket? I only found the closed topic from https://github.com/nuxsmin/sysPass/issues/500

regards, Mario

**Comment by nuxsmin** [comment]
Hi @vmario89 

This issue is related to a browser  security API issue, since it prevents to get copied any data into the clipboard from an asynchronous request.

The current implementation relies in a trick which retrieves the account's password and stores it in a hidden field, then it's copied into the clipboard. This process cannot take longer than X ms, because a restriction for user triggered events.

There is a browser API draft that would solve this issue.

Regards

**Comment by vmario89** [comment]
Hi, as i saw the issue is not solved for over 3 years now

https://github.com/nuxsmin/sysPass/issues/486
https://github.com/nuxsmin/sysPass/issues/158
https://github.com/nuxsmin/sysPass/issues/820
https://github.com/nuxsmin/sysPass/issues/486
https://github.com/nuxsmin/sysPass/issues/671

Do you have some idea how to prevent the browers to show up this warning? I am currently looking in browser configuration if there are some settings which modify the behaviour but no idea yet :-( 

regards, Mario

**Comment by vmario89** [comment]
i read, that Google announced using this API in Chrome 66 (https://developers.google.com/web/updates/2018/03/clipboardapi). Current version is Chrome 65. I guess the next update will be soon.

I also found out that "Copy to clipboard not working" does not appear on the following browsers:

- Internet Explorer v 11 (it asks the user about pasting to clipboard)
![grafik](https://user-images.githubusercontent.com/5631071/37612650-10fa5692-2ba6-11e8-8b14-5b55586bcc6c.png)
- FireFox Quantum v59
- WaterFox v56
- Vivald i1.14

Chrome and Edge are blocking without declaring the "problem" until now.

**Comment by nuxsmin** [comment]
@vmario89 that's interesting and would be a good starting point to solve this issue. I'd rather wait until all browsers have implemented it, since actually Chrome is the most used one (sadly from my point of view and Firefox's one...), and many users would complaint about it.

Thanks for the feedback!

Regards

**Comment by vmario89** [comment]
hey,
looking onto browser usage statistics this is really sad - i took a look into Wikipedia

its annoying for the 60% usership of Chrome, which is going to struggle with it

![grafik](https://user-images.githubusercontent.com/5631071/37924155-4894f95a-3131-11e8-88df-b22b6212c2c6.png)

**Comment by nuxsmin** [comment]
@vmario89 I'll analyze if this feature could be implemented seamlessly, so all the browsers could use it as soon as it becomes available for each one.

Thanks!

**Comment by Xumeiquer** [comment]
This is also present in V3. The server response is:
```Argument 1 passed to SP\Modules\Web\Controllers\Helpers\Account\AccountPasswordHelper::getPassword() must be an instance of SP\DataModel\AccountPassData, array given, called in /var/www/html/syspass/app/modules/web/Controllers/AccountController.php on line 578```

Hope this helps to solve the issue.

**Comment by vmario89** [comment]
Hi, The newest chrome version is 67+ now and it seems that the new api is online: https://medium.com/@harrisrobin/an-introduction-to-the-new-async-clipboard-api-7e6567685a05

**Comment by ortizleo** [comment]
Same problem here using last stable release 2.1 (2.1.16.18061901) with Firefox (63) and Chrome (70+).
Im using php 7.0. Can it be something with php and its modules?
